pkg/cmd/history_writer.go

```diff
@@ -20,17 +20,28 @@ import (
 )
 
 // Write writes history to given path
-func (w *GardenctlHistoryWriter) Write(historyPath string, history map[string]string) error {
+func (w *GardenctlHistoryWriter) Write(historyPath string, m map[string]string) error {
 	f, err := os.OpenFile(historyPath, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0666)
 	if err != nil {
 		return err
 	}
 
-	j, err := json.Marshal(history)
+	j, err := json.Marshal(m)
 	if err != nil {
 		return err
 	}
 
 	_, err = f.WriteString(string(j) + "\n")
 	return err
 }
+
+// WriteString writes history to given path
+func (w *GardenctlHistoryWriter) WriteString(historyPath string, s string) error {
```

Comment:
either remove the newline in `f.WriteString(s + "\n")` or rename this function to Writeln (similar to `fmt.Println`)

Reply:
refactor to `writeStringln`